### PR TITLE
network: add support for bridged bond to stage 2 kickstart (%pre)

### PR DESCRIPTION
**Pull request by rvykydal**
This is required to be able handle kickstart completely in stage 2.

### CODE REVIEW

**Review by poncovka — APPROVED** [review]
Looks good to me.

**Comment by rvykydal** [comment]
jenkins, test this please

**Comment by rvykydal** [comment]
I've added fixup for pylint which I'll squash before merging.

File: pyanaconda/modules/network/nm_client.py
Comment (jkonecny12) [inline pyanaconda/modules/network/nm_client.py:540]: I don't really like the `and` word in the function name. Most of the time it means that the functions should be split to two.

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Could you please squash these commits. The second does not really give benefit.

File: pyanaconda/modules/network/nm_client.py
Comment (rvykydal) [reply pyanaconda/modules/network/nm_client.py:540]: Why do you think this is the case ?
Also see: https://developer.gnome.org/libnm/stable/NMClient.html#nm-client-add-and-activate-connection-async

**Comment by rvykydal** [comment]
> Could you please squash these commits. The second does not really give benefit.

Please see the previous comment, it is for easier review for additional change for people that already reviewed the PR.

**Review by poncovka — APPROVED** [review]
Looks good to me.

File: pyanaconda/modules/network/nm_client.py
Comment (jkonecny12) [reply pyanaconda/modules/network/nm_client.py:540]: Because smaller function is easier to call and simpler for modification. Also having:
```
add_and_activate_connection_sync()
```
is not giving you much benefit compared to
```
add_connection_sync()
activate_connection_sync()
```

You will strip dependencies to just the parts requiring it. And the last reason is that it's simpler for testing because it's smaller.

However, this is not something for blocking this PR. It's just note. Reason for block was the commit squashing.

**Comment by jkonecny12** [comment]
> > Could you please squash these commits. The second does not really give benefit.
> 
> Please see the previous comment, it is for easier review for additional change for people that already reviewed the PR.

In general yes, but fixing pylint issues does not give any benefit to anyone. These are just changing how the code is written not what the code is doing so not really something requiring review. However, it will make harder to list history of the code and lines in the code basically it will shadow reason for the code change in the blame.

File: pyanaconda/modules/network/nm_client.py
Comment (VladimirSlavik) [reply pyanaconda/modules/network/nm_client.py:540]: If you wanted to get pedantic, it's used exactly once so maybe mark it as _private. Also, "activate" could be dropped from name since it's in the parameter list. But I don't think it's important...

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by rvykydal** [comment]
> > > Could you please squash these commits. The second does not really give benefit.
> > 
> > 
> > Please see the previous comment, it is for easier review for additional change for people that already reviewed the PR.
> 
> In general yes, but fixing pylint issues does not give any benefit to anyone. These are just changing how the code is written not what the code is doing so not really something requiring review. However, it will make harder to list history of the code and lines in the code basically it will shadow reason for the code change in the blame.

In my comment above I am just saying:
"I've added fixup for pylint which I'll squash before merging."

File: pyanaconda/modules/network/nm_client.py
Comment (rvykydal) [reply pyanaconda/modules/network/nm_client.py]: I split the function:
https://github.com/rhinstaller/anaconda/pull/2861/commits/ce593eed2b4706fd82876276b315c6010d5b8079

**Comment by jkonecny12** [comment]
> > > > Could you please squash these commits. The second does not really give benefit.
> > > 
> > > 
> > > Please see the previous comment, it is for easier review for additional change for people that already reviewed the PR.
> > 
> > 
> > In general yes, but fixing pylint issues does not give any benefit to anyone. These are just changing how the code is written not what the code is doing so not really something requiring review. However, it will make harder to list history of the code and lines in the code basically it will shadow reason for the code change in the blame.
> 
> In my comment above I am just saying:
> "I've added fixup for pylint which I'll squash before merging."

Ohh sorry my bad.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

**Pull request merged**
